### ISSUE TITLE
when using “icon-translate”, queryRenderedFeatures result unaffected/untranslated

### ISSUE THREAD

**Issue by livemixlove** [mapbox/mapbox-gl-js#2653]
(Moving from [stackexchange](http://stackoverflow.com/questions/37581950/mapboxgl-when-using-icon-translate-queryrenderedfeatures-result-unaffected), because it's seeming more like a bug.)

When “icon-translate” is set for an icon feature, queryRenderedFeatures returns the icon feature only when querying the original location.

Example:  open console to see you get results 20px below marker.
http://jsfiddle.net/k5y3n1yf/16/

(My workaround for now is to offset the query point for specific layers with offset markers, but this seems hackish.)

**Comment by livemixlove** [comment]
I see that "icon-offset" does what I want it to.  It still seems like "icon-translate" should affect queryRenderedFeatures results.

**Comment by lucaswoj** [comment]
I agree that `queryRenderedFeatures` should respect `icon-translate`. 

... I can also see why it doesn't by default. Our feature lookup is built on the worker thread after a `layout` property change but `icon-translate` is a paint property.

**Comment by jfirebaugh** [comment]
I believe that `*-translate` is handled for other layer types; the fact that `icon-translate` is not is probably just an oversight.

**Comment by strech345** [comment]
would it be solved in the next time?
